### DISCUSSION TITLE
MinimalLib RDKit build error

### DISCUSSION THREAD

**Discussion by syedzayyan** [rdkit/rdkit#7485] [Development]
Hello!

I was trying to build the MinimalLib and I constantly ran into a build fail with this error:

```
make[1]: *** [CMakeFiles/Makefile2:6904: Code/MinimalLib/CMakeFiles/RDKit_minimal.dir/rule] Error 2
make: *** [Makefile:2214: RDKit_minimal] Error 2
```

After a bit of digging, I found an error related to the rapidjson library. 

```
In file included from /home/syed-zayyan-masud/Desktop/rdkit/Code/GraphMol/MolInterchange/Parser.cpp:43:
/home/syed-zayyan-masud/Desktop/rdkit/External/rapidjson-1.1.0/include/rapidjson/document.h:319:82: error: cannot assign to non-static data member 'length' with const-qualified type 'const SizeType' (aka 'const unsigned int')
  319 |     GenericStringRef& operator=(const GenericStringRef& rhs) { s = rhs.s; length = rhs.length; }
      |                                                                           ~~~~~~ ^
```

Is this supposed to be normal?

I am on: 
Ubuntu 24.04, 
boost 1.84.0, 
gcc 13.2.0
latest RDKit (master)

Thanks

**Comment by tchaikov** [comment]
@syedzayyan it's a bug in rapidjson v1.1.0, and has been fixed in its master branch. unfortunately, ubuntu/jammy is not likely to receive an update of rapidjson package, not to mention that rapidjson has not released since v1.1.0 =(
